### ISSUE TITLE
[markdown] prettier breaks GitLab style math

### ISSUE THREAD

**Issue by ostrokach** [prettier/prettier#5798]
**Prettier 1.16.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEA1AhgJwJboEYA2cAzgAQAkABgB6XmnpQAmFlAnnQHQgA0IEABxjZoxZKCyYIAdwAKWBGJToAbhGxNeIPJnRgA1nBgBlAXuxQA5shiYArnD4ALGAFsCAdSfZ4xM2DhjRR9sFR82ZHBiMT4LYjhMGFldS1d0ZAAzdAJ4vgArYmoAIV0DI2N0VzgAGQs4TOzckALqYwtLIgBFOwh4BpzHEDNMeMxItMx9JhkoLQEcWA8NGCdkAA4ABj55iHiPXQFI+ZIElXq+TDgARztsS+T0VPSkLIG+eNdsG3tB4naunp9F6NQYwfBLJgrZAAJj4tnQ2AI7QAwhBXGlIlBoOcQHZ4gAVfBKV7xAC+pKAA)
```sh
--parser markdown
```

**Input:**
```markdown
Variables $`x`$ and $`y`$.
```

**Output:**
```markdown
Variables \$`x`$ and $`y`\$.

```

**Expected behavior:**

The line should be left unchanged because  "$\`x\`$ and $\`y\`$" are correct in [GitLab math](https://docs.gitlab.com/ee/user/markdown.html#math). I think it's better to leave the file unchanged than to correct some cases while breaking others?

**Comment by akraus53** [comment]
Any news on this? the error still occurs.

**Comment by peay** [comment]
@ikatyang any idea what could be done about this? In spite of https://github.com/prettier/prettier/issues/5425, this seems to be still happening in latest tag.

```
$`N_i`$, $`C_i`$  --> \$`N_i`$, $`C_i`\$ --> \$`N_i`\$, \$`C_i`\$  --> \$`N_i`\$, \\$`C_i`\$
```

and so on.

**Comment by thorn0** [comment]
Fixed by #7938
